server/data/data-jdbc/src/main/java/org/apache/james/mailrepository/jdbc/JDBCMailRepository.java

```diff
@@ -542,12 +544,18 @@ protected void internalStore(Mail mc) throws IOException, MessagingException {
                     insertMessage.setString(6, recipients.toString());
                     insertMessage.setString(7, mc.getRemoteHost());
                     insertMessage.setString(8, mc.getRemoteAddr());
-                    insertMessage.setTimestamp(9, new java.sql.Timestamp(mc.getLastUpdated().getTime()));
+                    if (mc.getPerRecipientSpecificHeaders().getHeadersByRecipient().isEmpty()) {
```

Comment:
is `mc.getPerRecipientSpecificHeaders()` could be null?

Reply:
As I recall, no it can't. Field is initialized in the ctor.